R/data-objects.R

```diff
@@ -56,24 +62,113 @@ iput <- function(
   stop_irods_overwrite(overwrite, logical_path)
 
   # what type of object are we dealing with
-  if (is.character(x) && file.exists(x)) {
-    fil <- x
-  } else if (is.symbol(substitute(x))) {
+  if (!(is.character(local_path) && file.exists(local_path))) {
+    stop("Local file does not exist.", call. = FALSE)
+  }
+
+  # handle file to iRODS conversion
+  out <-local_to_irods(
+    local_path = local_path,
+    logical_path = logical_path,
+    offset = offset,
+    count = count,
+    truncate = truncate,
+    verbose = verbose
+  )
+
+  invisible(out[[1]])
+}
+
+#' iRODS serialization Interface for Single R Objects
+#'
+#' Store an in-memory R object in iRODS with `isaveRDS()`. If the destination
+#' data-object or collection are not provided, the current iRODS collection and
+#' the input R object name are used. Get data-objects or collections from iRODS
+#' space in memory (R environment) with `ireadRDS()`.
+#'
+#' @param x R object stored on iRODS server.
+#' @param logical_path Destination path.
+#' @param offset Offset in bytes into the data object (Defaults to FALSE).
+#' @param truncate Truncates the object on open (defaults to TRUE).
+#' @param count Maximum number of bytes to read or write.
+#' @param verbose Show information about the http request and response.
+#' @param overwrite Overwrite irods object or local file (defaults to FALSE).
+#'
+#' @return Invisibly the http response in case of `isaveRDS()`, or an R object
+#' in case of `ireadRDS()`.
+#'
+#' @export
+#'
+#' @examples
+#' if(interactive()) {
+#' # connect project to server
+#' create_irods("http://localhost/irods-rest/0.9.3", "/tempZone/home")
+#'
+#' # authentication
+#' iauth()
+#'
+#' # store
+#' isaveRDS(iris, "iris.rds")
+#'
+#' # check if file is stored
+#' ils()
+#'
+#' # retrieve in native R format
+#' ireadRDS("iris.rds")
+#' }
+isaveRDS <- function(
+    x,
+    logical_path,
+    offset = 0,
+    count = 2000L,
+    truncate = "true",
+    verbose = FALSE,
+    overwrite = FALSE
+) {
+
+  # expand logical path to absolute logical path
+  logical_path <- get_absolute_lpath(logical_path)
+
+  # check if irods object already exists and whether it should be overwritten
+  stop_irods_overwrite(overwrite, logical_path)
+
+  # serialize R object
+  if (is.symbol(substitute(x))) {
     # object name
     name <- deparse(substitute(x))
     # # make intermediate file
     fil <- tempfile(name, fileext = ".rds")
     # serialize and gz compress
     saveRDS(x, fil)
   } else {
-    stop("Local object or file does not exist.", call. = FALSE)
+    stop("Local object does not exist.", call. = FALSE)
```

Comment:
could name "[x]" here to help user?  not sure the convention for error messages in R.

Reply:
I think it's a good idea:
``stop(sprintf("Local object `%s` does not exist", deparse(substitute(x))), call. = FALSE)``.

By the way `is.symbol(substitute(x))` could be replaced with `exists(x)`.

Reply:
already did that, only have to push that again.

Reply:
I think I found that the problems originate from this. I also thought that this would be a better way, using `exists()`. And it is, but I forgot to add the environment in which to search. It should be `exists("<object name>", envir = envir = parent.frame()) `.